ledger/acctupdates.go

```diff
@@ -1380,6 +1379,29 @@ func (au *accountUpdates) commitRound(ctx context.Context, tx *sql.Tx, dcc *defe
 		return err
 	}
 
+	addressesNeeded := dcc.compactResourcesDeltas.getMissingAddresses()
+	addressesResolved := make(map[basics.Address]int64, len(addressesNeeded))
+	for _, delta := range dcc.compactAccountDeltas.deltas {
+		if _, ok := addressesNeeded[delta.oldAcct.addr]; ok {
+			addressesResolved[delta.oldAcct.addr] = delta.oldAcct.rowid
+		}
+	}
+
+	au.accountsMu.RLock()
+	for addr := range addressesNeeded {
+		if _, ok := addressesResolved[addr]; !ok {
+			if pad, has := au.baseAccounts.read(addr); has {
+				addressesResolved[pad.addr] = pad.rowid
+			}
+		}
+	}
+	au.accountsMu.RUnlock()
```

Comment:
this portion is correct, but need to be moved to the `prepareCommit`, since the lock is already taken there. The compactResourcesDeltas should also contain the resolved addresses so that you won't need to pass it over the dcc.